app/models/user.rb

```diff
@@ -75,4 +75,19 @@ def requires_no_invitation_code!
   def requires_no_invitation_code?
     !!@requires_no_invitation_code
   end
+
+  def undfined_agent_types
+    agents.unscoped.group(:type).pluck(:type).select do |type|
+      begin
+        type.constantize
+        false
+      rescue NameError
+        true
+      end
+    end
+  end
+
+  def undefined_agents
+    agents.unscoped.where(type: undfined_agent_types).select('id, schedule, type as undefined')
```

Comment:
It's the only way I know to prevent the STI which would then raise `ActiveRecord::SubclassNotFound` again.